Packs/MISP/Integrations/MISPV3/MISPV3.py

```diff
@@ -0,0 +1,1436 @@
+        raw_response=updated_event
+    )
+
+
+def generic_reputation_command(demisto_args, reputation_type, dbot_type, malicious_tag_ids, suspicious_tag_ids,
+                               reliability):
+    reputation_value_list = argToList(demisto_args.get(reputation_type), ',')
+    command_results = []
+    for value in reputation_value_list:
+        command_results.append(
+            get_indicator_results(value, dbot_type, malicious_tag_ids, suspicious_tag_ids, reliability))
+    return command_results
+
+
+def reputation_value_validation(value, dbot_type):
+    if dbot_type == 'FILE':
+        # hashFormat will be used only in output
+        hash_format = get_hash_type(value)
+        if hash_format == 'Unknown':
+            raise DemistoException('Invalid hash length, enter file hash of format MD5, SHA-1 or SHA-256')
+    if dbot_type == 'IP':
+        if not is_ip_valid(value):
+            raise DemistoException(f"Error: The given IP address: {value} is not valid")
+    if dbot_type == 'DOMAIN':
+        if not re.compile(DOMAIN_REGEX, regexFlags).match(value):
+            raise DemistoException(f"Error: The given domain: {value} is not valid")
+    if dbot_type == 'URL':
+        if not re.compile(urlRegex, regexFlags).match(value):
+            raise DemistoException(f"Error: The given url: {value} is not valid")
+    if dbot_type == 'EMAIL':
+        if not re.compile(emailRegex, regexFlags).match(value):
+            raise DemistoException(f"Error: The given email address: {value} is not valid")
+
+
+def get_indicator_results(value, dbot_type, malicious_tag_ids, suspicious_tag_ids, reliability):
+    """
+    This function searches for the given attribute value in MISP and then calculates it's dbot score.
+    The score is calculated by the tags ids (attribute tags and event tags).
+    Args:
+        value (str): The indicator value (an IP address, email address, domain, url or file hash).
+        dbot_type (str): Indicator type (file, url, domain, email or ip).
+        malicious_tag_ids (set): Tag ids should be recognised as malicious.
+        suspicious_tag_ids (set): Tag ids should be recognised as suspicious
+        reliability (DBotScoreReliability): integration reliability score.
+
+    Returns:
+        CommandResults includes all the indicator results.
+    """
+    reputation_value_validation(value, dbot_type)
+    misp_response = PYMISP.search(value=value, controller='attributes', include_context=True,
+                                  include_correlations=True, include_event_tags=True, enforce_warninglist=True,
+                                  include_decay_score=True, includeSightings=True)
+    indicator_type = INDICATOR_TYPE_TO_DBOT_SCORE[dbot_type]
+    is_indicator_found = misp_response and misp_response.get('Attribute')
+    if is_indicator_found:
+        outputs, score, found_tag, found_related_events = parse_response_reputation_command(
+            copy.deepcopy(misp_response),
```

Comment:
it should be in the function responsibility to copy the argument(in order to protect it) and not the caller.